### PR TITLE
Fix sets landing page

### LINKED ISSUE #703: Clicking facets link gives error when on oregondigital.org/sets
When on the Sets list page, http://oregondigital.org/sets  clicking facets gives the error "We're sorry, but something went wrong."

The error in the logs is: 
ActionView::Template::Error (No route matches {:action=>"show", :controller=>"sets", :id=>"oregondigital:pg15bf25q", :set=>nil})

coming via link_to_document() in collections_helper.rb

### PR DESCRIPTION
**Pull request by jechols**
Fixes #703

Facets are dropped entirely, and the search box properly links to the
catalog when there's no selected set

### CODE REVIEW

**Comment by wickr** [comment]
I think the sets landing page should still show the facets.

**Comment by jechols** [comment]
Mentioned it in slack but not here - this is probably not "the" solution.  More a reason to get discussions going.  The bug definitely needs a fix, and this fixes it, but yeah, we need to figure out what we are okay with, how much time to put into fixing this properly, etc.

**Comment by wickr** [comment]
Despite #794, I took a quick look at this and think I found a fix. Pushing new commits here now.

**Comment by terrellt** [comment]
> Despite #794, I took a quick look at this and think I found a fix. Pushing new commits here now.

Can we get a test to prove this works?

File: app/controllers/sets_controller.rb [hunk 1]
```diff
@@ -14,7 +14,8 @@ def index
   end
 
   def search_action_url(options={})
-    sets_path(options.merge(:set => @collection))
+    return sets_path(options.merge(:set => @collection)) if @collection
+    return root_path
```
Comment (terrellt) [inline app/controllers/sets_controller.rb]: Implicit return please. Feel free to skip CI

**Pull request merged**